File: backend/entity/Portfolio.go
package entity

import (
	"gorm.io/gorm"
)

type Portfolio struct {
	gorm.Model
	Title             string           `gorm:"unique" json:"title"`
	Description       string           `json:"description"`
	PortType          string           `json:"porttype"`
	LinkPortfolio     string           `json:"link_portfolio"`
	File_urls         string           `json:"file_urls"`
	UserID            uint             `gorm:"not null" json:"user_id"`
	AdminComment      *string          `json:"admin_comment"`
	PortfolioStatusID uint             `gorm:"not null" json:"portfolio_status_id"`
	PortfolioStatus   *PortfolioStatus `gorm:"foreignKey:PortfolioStatusID" json:"portfolio_status"`
}

File: frontend/src/interfaces/portfolio.ts
// interfaces/Portfolio.ts
import type { PortfolioStatusInterface } from "./portfoliostatus";

export interface PortfolioInterface {
  ID?: number;
  title?: string;
  description?: string;
  porttype?: string; 
  link_portfolio?: string;
  file_urls?: string;
  user_id?: number;
  admin_comment?: string | null;
  portfolio_status_id?: number;
  portfolio_status?: PortfolioStatusInterface;

  CreatedAt?: string;
  UpdatedAt?: string;
  DeletedAt ?: string | null;
}